src/app-router.js

```diff
@@ -28,7 +28,11 @@ export class AppRouter extends Router {
   reset() {
     super.reset();
     this.maxInstructionCount = 10;
-    this._queue = [];
+    if (!this._queue) {
+      this._queue = [];
```

Comment:
I could refactor it. I don't like putting initialization code in a reset function. Is that okay?